src/panels/detail/DetailLink.tsx

```diff
@@ -81,46 +82,50 @@ export default class DetailLink extends React.Component<Props, State> {
         return result;
     }
 
-    prepareDetails(id: string) {
-        let sourceCard = ProjectLinks[id].sourceCardinality;
-        let targetCard = ProjectLinks[id].targetCardinality;
-        this.setState({
-            sourceCardinality: "0",
-            targetCardinality: "0"
-        });
-        CardinalityPool.forEach((card, i) => {
-            if (sourceCard.getString() === card.getString()) {
-                this.setState({sourceCardinality: i.toString(10)});
-            }
-            if (targetCard.getString() === card.getString()) {
-                this.setState({targetCardinality: i.toString(10)});
-            }
-        })
-        this.setState({
-            iri: ProjectLinks[id].iri,
-            changes: false,
-            readOnly: Schemes[VocabularyElements[ProjectElements[ProjectLinks[id].source].iri].inScheme].readOnly
-        });
+    prepareDetails(id?: string) {
+        if (id) {
+            let sourceCard = ProjectLinks[id].sourceCardinality;
+            let targetCard = ProjectLinks[id].targetCardinality;
+            this.setState({
+                sourceCardinality: "0",
+                targetCardinality: "0"
+            });
+            CardinalityPool.forEach((card, i) => {
+                if (sourceCard.getString() === card.getString()) {
+                    this.setState({sourceCardinality: i.toString(10)});
+                }
+                if (targetCard.getString() === card.getString()) {
+                    this.setState({targetCardinality: i.toString(10)});
+                }
+            })
+            this.setState({
+                id: id,
+                iri: ProjectLinks[id].iri,
+                changes: false,
+                readOnly: Schemes[VocabularyElements[ProjectElements[ProjectLinks[id].source].iri].inScheme].readOnly
+            });
```

Comment:
The 2 - 4 `this.setState` calls should be grouped to one. As far as I can tell the current code would trigger multiple React component updates. So I'd rather create some temp state object and set it using a single setState call. There are ways to batch state updates, but none that I know are applicable to this code.